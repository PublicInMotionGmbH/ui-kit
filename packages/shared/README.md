# Talixo Shared

Shared data between whole UI Kit.

## How to install

Package is available as `@talixo/shared` in NPM registry, so you can use it in your project
using `npm install @talixo/shared --save` or `yarn add @talixo/shared`.

## Requirements

Your package should additionally have some extra dependencies:

- `classnames: ^2.2.5`

These packages are required by `@talixo/shared`, but you have to install them manually,
to avoid having different versions of these in your application.
