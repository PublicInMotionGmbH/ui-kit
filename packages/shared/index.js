import applyAnyClassNameModifiers from './src/applyAnyClassNameModifiers'
import applyClassNameModifiers from './src/applyClassNameModifiers'
import buildClassName from './src/buildClassName'
import prefix from './src/prefix'

export {
  applyAnyClassNameModifiers,
  applyClassNameModifiers,
  buildClassName,
  prefix
}
