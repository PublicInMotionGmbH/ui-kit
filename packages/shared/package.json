{
  "name": "@talixo/shared",
  "version": "1.0.0",
  "description": "Talixo common configuration",
  "author": "Public in Motion GmbH",
  "main": "index.js",
  "module": "index.js",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "https://github.com/PublicInMotionGmbH/ui-kit.git"
  },
  "bugs": {
    "url": "https://github.com/PublicInMotionGmbH/ui-kit/issues"
  },
  "devDependencies": {
    "rollup": "^0.57.1"
  },
  "dependencies": {
    "normalize.css": "^8.0.0"
  }
}
