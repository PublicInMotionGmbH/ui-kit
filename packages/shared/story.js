import { storiesOf } from '@storybook/react'
import { withInfo } from '@storybook/addon-info'

import createController from './story/createController'

/**
 * Build styles for story
 *
 * @param {object} stylesheet
 * @param {object} options
 * @returns {object}
 */
const styles = (stylesheet, options) => ({
  ...stylesheet,
  infoBody: {
    ...stylesheet.infoBody,
    boxShadow: 'none',
    border: 0,
    padding: 0
  },
  propTableHead: options.propTables && options.propTables.length > 1 ? {

  } : {
    display: 'none'
  },
  source: {
    ...stylesheet.source,
    h1: {
      ...stylesheet.source.h1,
      marginBottom: '15px'
    }
  }
})

/**
 * Create factory for stories
 *
 * @param {string} name
 * @param {object} module
 * @param {object} defaultOptions
 *
 * @returns {function(name: string, description: string, render: function)|function(name: string, description: string, render: function, infoOptions: object)|{ controlled: function(name: string, description: string, render: function, getInitialState: function, infoOptions: object)|function(name: string, description: string, render: function, getInitialState: function, infoOptions: object) }}
 */
export function createStoriesFactory (name, module, defaultOptions = {}) {
  const stories = storiesOf(name, module)

  function addStory (name, description, render, infoOptions = {}) {
    // When rendering function expect any arguments,
    // It means that it's controlled
    if (render.length > 0) {
      render = createController(render)

      infoOptions = {
        ...infoOptions,
        propTablesExclude: [
          render.Controller,
          ...(infoOptions.propTablesExclude || [])
        ]
      }
    }

    // Build options for story
    const options = {
      header: true,
      inline: true,
      text: description,
      ...defaultOptions,
      ...infoOptions
    }

    // Build styles factory
    const stylesFunction = options.styles
      ? stylesheet => options.styles(styles(stylesheet, options), options)
      : stylesheet => styles(stylesheet, options)

    // Create story
    return stories.add(name, withInfo({
      ...options,
      styles: stylesFunction
    })(render))
  }

  return addStory
}

/**
 * Get description from README file
 *
 * @param {string} content
 * @returns {string}
 */
export function getReadmeDescription (content) {
  return content
    .replace(/<p><h1([^>]*)>(.*)<\/h1><\/p>/g, '<h1$1>$2</h1>')
    .replace(/^<h1[^>]*>.*<\/h1>/, '')
    .replace(/(<p>)?<h[1-6][> ][^]*$/i, '')
}
