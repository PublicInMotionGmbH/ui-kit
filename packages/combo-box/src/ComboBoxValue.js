import React from 'react'
import PropTypes from 'prop-types'

import { buildClassName } from '@talixo/shared'

import { Icon } from '@talixo/icon'

const propTypes = {
  /** Function to build toggle props with handlers */
  getToggleButtonProps: PropTypes.func.isRequired,

  /** Function to build clear button props with handlers */
  getClearButtonProps: PropTypes.func.isRequired,

  /** Function to build input props with handlers */
  getInputProps: PropTypes.func.isRequired,

  /** Function to render item value */
  renderValue: PropTypes.func.isRequired,

  /** Value provided to input */
  inputValue: PropTypes.string,

  /** Additional icon to put in the select box */
  icon: PropTypes.node,

  /** Is it multi-select? */
  multi: PropTypes.bool,

  /** Placeholder to show when there is no value */
  placeholder: PropTypes.string,

  /** Is menu open? */
  isOpen: PropTypes.bool,

  /** List of currently selected items */
  selectedItems: PropTypes.array
}

const defaultProps = {
  isOpen: false,
  selectedItems: []
}

/**
 * Value shown in single-select ComboBox.
 *
 * @property {object} props
 * @property {function} props.getToggleButtonProps
 * @property {function} props.getClearButtonProps
 * @property {function} props.getInputProps
 * @property {function} props.renderValue
 * @property {boolean} props.multi
 * @property {boolean} props.isOpen
 * @property {array} props.selectedItems
 * @property {string} [props.placeholder]
 * @property {*} [props.icon]
 *
 * @property {*} focusTimeout
 * @property {HTMLElement|null} [input]
 *
 * @class
 */
class ComboBoxValue extends React.PureComponent {
  /**
   * Cancel delayed focus on input,
   * when component is unmounted anyway.
   */
  componentWillUnmount () {
    this.cancelDelayedFocus()
  }

  /**
   * Save reference to input for further use.
   *
   * @param {HTMLElement} el
   */
  saveInputRef = (el) => {
    this.input = el
  }

  /**
   * Focus input if it is possible.
   */
  focus = () => {
    if (this.input) {
      this.input.focus()
    }
  }

  /**
   * Focus input if it is possible (on next tick).
   */
  delayedFocus = () => {
    this.focusTimeout = setTimeout(() => this.focus(), 1)
  }

  /**
   * Cancel delayed focus on input.
   */
  cancelDelayedFocus () {
    clearTimeout(this.focusTimeout)
  }

  /**
   * Build 'clear' button which can remove current selection.
   *
   * @returns {React.Element}
   */
  buildClearButton () {
    const { getClearButtonProps } = this.props

    // Build class name for 'clear' button
    const clearClsName = buildClassName([ 'combo-box', 'clear' ])

    // Build 'clear' button props,
    // including focus on input after clearing
    const clearProps = getClearButtonProps({
      role: 'button',
      className: clearClsName,
      onClick: this.delayedFocus
    })

    return (
      <span {...clearProps}><Icon name='clear' /></span>
    )
  }

  /**
   * Build arrow which determines a state of list.
   *
   * @returns {React.Element}
   */
  buildArrow () {
    const { isOpen } = this.props

    // Build class name for arrow element
    const arrowClsName = buildClassName([ 'combo-box', 'arrow' ])

    return (
      <span className={arrowClsName}>
        <Icon name={isOpen ? 'expand_less' : 'expand_more'} />
      </span>
    )
  }

  /**
   * Build controls on end of combo-box.
   * By default it is arrow which shows list state,
   * but it can be 'clear' button as well (when element is select).
   *
   * Additionally, there can be external icon passed.
   *
   * @returns {React.Element}
   */
  buildControls () {
    const { icon, selectedItems } = this.props

    // Get information about selection
    const isSelected = selectedItems.length > 0

    // Build class names for element
    const controlsClsName = buildClassName([ 'combo-box', 'controls' ])
    const iconClsName = buildClassName([ 'combo-box', 'icon' ])

    // Build element for external icom
    const iconElement = icon ? <div className={iconClsName}>{icon}</div> : null

    // Build main control which should be shown on end
    const mainControl = isSelected
      ? this.buildClearButton()
      : this.buildArrow()

    return (
      <div className={controlsClsName}>
        {iconElement}
        {mainControl}
      </div>
    )
  }

  /**
   * Build text input to provide data for combo-box.
   *
   * @returns {React.Element}
   */
  buildInput () {
    const { getInputProps, placeholder, inputValue } = this.props

    // Build props for input
    const inputProps = getInputProps({
      type: 'text',
      placeholder: placeholder,
      value: inputValue
    })

    return (
      <input ref={this.saveInputRef} {...inputProps} />
    )
  }

  /**
   * Render value box for single-select combo-box.
   *
   * @returns {React.Element}
   */
  render () {
    const { getToggleButtonProps, selectedItems, inputValue, renderValue } = this.props

    // Get information about selection
    const isSelected = selectedItems.length > 0

    // Get properties required by toggle button
    const passedProps = getToggleButtonProps({
      onFocus: this.focus,
      onClick: this.focus
    })

    // Build class names for elements
    const innerClsName = buildClassName([ 'combo-box', 'inner' ])
    const clsName = buildClassName([ 'combo-box', 'value' ], null, {
      placeholder: isSelected ? false : inputValue == null || inputValue === ''
    })

    // Build either input for new value or show current value
    const value = isSelected
      ? renderValue(selectedItems[0], this.props)
      : this.buildInput()

    // Render combo-box value container
    return (
      <button {...passedProps} className={clsName}>
        <div className={innerClsName}>
          {value}
        </div>
        {this.buildControls()}
      </button>
    )
  }
}

ComboBoxValue.displayName = 'ComboBoxValue'

ComboBoxValue.propTypes = propTypes
ComboBoxValue.defaultProps = defaultProps

export default ComboBoxValue
