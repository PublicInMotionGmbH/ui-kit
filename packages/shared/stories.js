import React from 'react'
import { createStoriesFactory } from './story'

// Load first paragraph from README file
const readme = ''

// Create factories for story
const addStory = createStoriesFactory('Typography and basics', module)

// Stories

addStory('Headers', readme, () => (
  <div>
    <h1>h1. Header</h1>
    <h2>h2. Header</h2>
    <h3>h3. Header</h3>
    <h4>h4. Header</h4>
    <h5>h5. Header</h5>
    <h6>h6. Header</h6>

    <h1>h1. Header <small>Secondary text</small></h1>
    <h2>h2. Header <small>Secondary text</small></h2>
    <h3>h3. Header <small>Secondary text</small></h3>
    <h4>h4. Header <small>Secondary text</small></h4>
    <h5>h5. Header <small>Secondary text</small></h5>
    <h6>h6. Header <small>Secondary text</small></h6>
  </div>
))

addStory('Text styling', readme, () => (
  <div>
    <h2>Paragraphs</h2>

    <p>
      Lorem ipsum dolor sit amet enim. Etiam ullamcorper. Suspendisse a pellentesque dui, non felis.
      Maecenas malesuada elit lectus felis, malesuada ultricies.
      Curabitur et ligula. Ut molestie a, ultricies porta urna.
      Vestibulum commodo volutpat a, convallis ac, laoreet enim. Phasellus fermentum in, dolor.
    </p>

    <p>
      Pellentesque facilisis. Nulla imperdiet sit amet magna.
      Vestibulum dapibus, mauris nec malesuada fames ac turpis velit, rhoncus eu, luctus et interdum adipiscing wisi.
    </p>

    <p>
      Aliquam erat ac ipsum. Integer aliquam purus.
      Quisque lorem tortor fringilla sed, vestibulum id, eleifend justo vel bibendum sapien massa ac turpis
      faucibus orci luctus non, consectetuer lobortis quis, varius in, purus.
      Integer ultrices posuere cubilia Curae, Nulla ipsum dolor lacus, suscipit adipiscing.
      Cum sociis natoque penatibus et ultrices volutpat.
    </p>

    <h2>Inline styling</h2>

    <p>
      We may have some <strong>strong elements</strong> inside our text, as well as some <em>emphasises</em>.
      When we want to calculate 2<sup>3</sup> + x<sub>1</sub>, we may have problems, but it's easy to show it.
      <br />
      In some circumstances, mostly describing our API or other stuff we may also try to show some <code>short code snippet</code>.
      It happens that we may also use <kbd>Ctrl+C</kbd> to hint user with keyboard shortcuts.
    </p>

    <p>
      It may happen that we would like to <mark>highlight</mark> some content inside our dashboards, in case of some search engine.
    </p>

    <p>
      <del>We do not make any mistakes.</del> When we do, we can write it semantically marking that anyway.
    </p>

    <p>
      When we use some shortcuts like <abbr title='Looks good to me'>LGTM</abbr>, we can show such information for user.
    </p>

    <h2>Quotes and testimonials</h2>

    <blockquote>
      This was one of the best experiences I have ever had with a cab company.
      I had problems at the airport and the driver stayed with me for over an hour and helped me sort everything out.
      I would recommend this company to anyone. Thank you for such fabulous service!
    </blockquote>

    <blockquote>
      This was one of the best experiences I have ever had with a cab company.
      I had problems at the airport and the driver stayed with me for over an hour and helped me sort everything out.
      I would recommend this company to anyone. Thank you for such fabulous service!

      <footer>
        <cite>R.M.</cite>
      </footer>
    </blockquote>
  </div>
))

addStory('RTL: Headers', readme, () => (
  <div dir='rtl'>
    <h1>h1. כּוֹתֶרֶת</h1>
    <h2>h2. כּוֹתֶרֶת</h2>
    <h3>h3. כּוֹתֶרֶת</h3>
    <h4>h4. כּוֹתֶרֶת</h4>
    <h5>h5. כּוֹתֶרֶת</h5>
    <h6>h6. כּוֹתֶרֶת</h6>

    <h1>h1. כּוֹתֶרֶת <small>טקסט משני</small></h1>
    <h2>h2. כּוֹתֶרֶת <small>טקסט משני</small></h2>
    <h3>h3. כּוֹתֶרֶת <small>טקסט משני</small></h3>
    <h4>h4. כּוֹתֶרֶת <small>טקסט משני</small></h4>
    <h5>h5. כּוֹתֶרֶת <small>טקסט משני</small></h5>
    <h6>h6. כּוֹתֶרֶת <small>טקסט משני</small></h6>
  </div>
))

addStory('RTL: Text styling', readme, () => (
  <div dir='rtl'>
    <h2>פסקאות</h2>

    <p>
      אחד אל בדפים חרטומים, או פיסול צרפתית ויקיפדיה היא, אם מדריכים הגולשות כתב. ב עוד המחשב ברוכים עקרונות. ספינות מדריכים אקטואליה אחר על.
      זאת אם ביולוגיה אדריכלות, עוד ראשי וקשקש מתמטיקה ב. ויש בה רביעי תרומה לציין. זקוק המדינה גם עוד.
      זאת דת לתרום הבאים נוסחאות, אם פיסיקה סטטיסטיקה אווירונאוטיקה רבה.
    </p>

    <p>
      נפלו למנוע מונחונים או כדי. אחרות תקשורת בה אחר. מיזם יסוד רביעי בה מלא, הבקשה בדפים על זכר. כלל או העזרה ממונרכיה, כלל גם הטבע אנציקלופדיה, אנא אל שמות יסוד.
    </p>

    <p>
תיקונים והגולשים אל בדף, של בדף בישול יוצרים העברית. עזה אם והוא הארץ בידור, תחבורה משופרות עזה אם.
      מה רוסית בארגז הגולשות צ'ט, קבלו מוסיקה ב אחד, מדעי אדריכלות בדף בה. מדע שתפו ויקימדיה אם.
      אדריכלות אווירונאוטיקה אתה של, תרבות ולחבר צ'ט דת.
      לחשבון לויקיפדיה עזה אל, תנך דרכה בכפוף מיוחדים של.
      שתי מה רפואה אודות, כיצד רקטות סטטיסטיקה דת שמו. עוד אקראי וקשקש מיוחדים אם,
      לוח גם העזרה בהתייחסות. מה מונחים אספרנטו קרן. אינו לעריכה של חפש, בה מאמר מוגש אנא.
      בקר הרוח מתמטיקה את, בה החלה יסוד הגולשות קרן. בה לימודים ממונרכיה ואלקטרוניקה עזה,
      חופשית מועמדים מדע של. מדע עזרה אחרות המלצת של, דת יוני אנגלית כתב.
    </p>

    <h2>עיצוב מוטבע</h2>

    <p>
      אולי יש לנו<strong> כמה אלמנטים חזקים בתוך הטקסט </strong>שלנו, כמו גם כמה מדגיש.
      ב במקרים מסוימים, בעיקר לתאר את ה- API שלנו או דברים אחרים אנו עשויים<code> גם לנסות להציג קטע קוד קצר. זה קורה</code>כי אנו יכולים גם >להשתמ
      <kbd>Ctrl+C</kbd> לרמז למשתמש עם קיצורי מקשים.
    </p>

    <p>
      זה יכול לקרות כי אנחנו רוצים להדגיש קצת תוכן בתוך לוחות.<mark> המחוונים שלנו, במקרה של מנוע </mark>החיפוש
    </p>

    <p>
      <del>אנחנו לא עושים שום טעויו</del>ת. כאשר אנו עושים, אנחנו יכולים לכתוב את זה סמנטית סימון כי בכל מקרה.
    </p>

    <p>
      כאשר אנו משתמשים בקיצורי דרך מסוימים כמו <abbr title='נראה לי טוב'>LGTM</abbr>, אנו יכולים להציג מידע כזה עבור המשתמש.
    </p>

    <h2>ציטוטים ועדויות</h2>

    <blockquote>
      זה היה אחד החוויות הטובות ביותר שהיו לי אי פעם עם חברת מוניות.
היו לי בעיות בשדה התעופה והנהג נשאר איתי יותר משעה ועזר לי למיין הכול.
אני ממליץ על חברה זו לכל אחד. תודה לך על שירות נהדר
    </blockquote>

    <blockquote>
      זה היה אחד החוויות הטובות ביותר שהיו לי אי פעם עם חברת מוניות.
היו לי בעיות בשדה התעופה והנהג נשאר איתי יותר משעה ועזר לי למיין הכול.
אני ממליץ על חברה זו לכל אחד. תודה לך על שירות נהדר
      <footer>
        <cite>מה-שמו</cite>
      </footer>
    </blockquote>

    <div style={{ display: 'flex' }}>
      <blockquote style={{ marginLeft: '15px' }}>
        זה היה אחד החוויות הטובות ביותר שהיו לי אי פעם עם חברת מוניות.
  היו לי בעיות בשדה התעופה והנהג נשאר איתי יותר משעה ועזר לי למיין הכול.
  אני ממליץ על חברה זו לכל אחד. תודה לך על שירות נהדר
        <footer>
          <cite>מה-שמו</cite>
        </footer>
      </blockquote>

      <blockquote>
        זה היה אחד החוויות הטובות ביותר שהיו לי אי פעם עם חברת מוניות.
  היו לי בעיות בשדה התעופה והנהג נשאר איתי יותר משעה ועזר לי למיין הכול.
  אני ממליץ על חברה זו לכל אחד. תודה לך על שירות נהדר
        <footer>
          <cite>מה-שמו</cite>
        </footer>
      </blockquote>
    </div>
  </div>
))
