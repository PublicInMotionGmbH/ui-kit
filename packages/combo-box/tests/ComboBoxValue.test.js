import React from 'react'
import { shallow, mount } from 'enzyme'

import { prefix } from '@talixo/shared'

import ComboBoxValue from '../src/ComboBoxValue'

const moduleName = prefix('combo-box', 'value')
const innerName = prefix('combo-box', 'inner')
const clearName = prefix('combo-box', 'clear')
const arrowName = prefix('combo-box', 'arrow')

describe('<ComboBoxValue />', () => {
  beforeEach(() => jest.useFakeTimers())
  afterEach(() => jest.useRealTimers())

  it('should render correctly', () => {
    const wrapper = shallow(
      <ComboBoxValue
        getToggleButtonProps={() => {}}
        getInputProps={() => {}}
        getClearButtonProps={() => {}}
        renderValue={item => item}
      />
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('should not add information about placeholder to wrapper', () => {
    const wrapper = shallow(
      <ComboBoxValue
        selectedItems={[ 'abc' ]}
        getToggleButtonProps={() => {}}
        getInputProps={() => {}}
        getClearButtonProps={() => {}}
        renderValue={item => item}
      />
    )

    expect(wrapper.hasClass(`${moduleName}--placeholder`)).toBe(false)
  })

  it('should add information about placeholder to wrapper', () => {
    const wrapper = shallow(
      <ComboBoxValue
        getToggleButtonProps={() => {}}
        getInputProps={() => {}}
        getClearButtonProps={() => {}}
        renderValue={item => item}
      />
    )

    expect(wrapper.hasClass(`${moduleName}--placeholder`)).toBe(true)
  })

  it('should use getToggleButtonProps', () => {
    const wrapper = shallow(
      <ComboBoxValue
        getToggleButtonProps={() => ({ role: 'test' })}
        getInputProps={() => {}}
        getClearButtonProps={() => {}}
        renderValue={item => item}
      />
    )

    expect(wrapper.prop('role')).toBe('test')
  })

  it('should use getInputProps', () => {
    const wrapper = shallow(
      <ComboBoxValue
        getToggleButtonProps={() => {}}
        getInputProps={() => ({ role: 'test' })}
        getClearButtonProps={() => {}}
        renderValue={item => item}
      />
    )

    expect(wrapper.find(`.${innerName}`).find('input').prop('role')).toBe('test')
  })

  it('should use getClearButtonProps', () => {
    const wrapper = shallow(
      <ComboBoxValue
        selectedItems={[ 'abc' ]}
        getToggleButtonProps={() => {}}
        getInputProps={() => {}}
        getClearButtonProps={props => ({ ...props, role: 'test' })}
        renderValue={item => item}
      />
    )

    expect(wrapper.find(`.${clearName}`).prop('role')).toBe('test')
  })

  it('should render input when there is no value', () => {
    const wrapper = shallow(
      <ComboBoxValue
        getToggleButtonProps={() => {}}
        getInputProps={() => {}}
        getClearButtonProps={() => {}}
        renderValue={item => item}
      />
    )

    expect(wrapper.find(`.${innerName}`).find('input').length).toBe(1)
  })

  it('should render input with value', () => {
    const wrapper = shallow(
      <ComboBoxValue
        inputValue='abc'
        getToggleButtonProps={() => {}}
        getInputProps={props => props}
        getClearButtonProps={() => {}}
        renderValue={item => item}
      />
    )

    expect(wrapper.find(`.${innerName}`).find('input').prop('value')).toBe('abc')
  })

  it('should render value when there is selected', () => {
    const wrapper = shallow(
      <ComboBoxValue
        selectedItems={[ 'abc' ]}
        getToggleButtonProps={() => {}}
        getInputProps={() => {}}
        getClearButtonProps={() => {}}
        renderValue={item => item}
      />
    )

    expect(wrapper.find(`.${innerName}`).text()).toBe('abc')
  })

  it('should render special value when there is selected', () => {
    const wrapper = shallow(
      <ComboBoxValue
        selectedItems={[ { id: 'abc' } ]}
        getToggleButtonProps={() => {}}
        getInputProps={() => {}}
        getClearButtonProps={() => {}}
        renderValue={item => <span id='test'>{item.id}</span>}
      />
    )

    expect(wrapper.find(`.${innerName}`).find('#test').length).toBe(1)
    expect(wrapper.find(`.${innerName}`).find('#test').text()).toBe('abc')
  })

  it('should show icon when it is passed', () => {
    const wrapper = shallow(
      <ComboBoxValue
        icon={<span id='test' />}
        getToggleButtonProps={() => {}}
        getInputProps={() => {}}
        getClearButtonProps={() => {}}
        renderValue={item => item}
      />
    )

    expect(wrapper.find(`#test`).length).toBe(1)
  })

  it('should show open arrow', () => {
    const wrapper = shallow(
      <ComboBoxValue
        isOpen={false}
        getToggleButtonProps={() => {}}
        getInputProps={() => {}}
        getClearButtonProps={() => {}}
        renderValue={item => item}
      />
    )

    expect(wrapper.find(`.${arrowName}`).find('Icon').prop('name')).toBe('expand_more')
  })

  it('should show close arrow', () => {
    const wrapper = shallow(
      <ComboBoxValue
        isOpen
        getToggleButtonProps={() => {}}
        getInputProps={() => {}}
        getClearButtonProps={() => {}}
        renderValue={item => item}
      />
    )

    expect(wrapper.find(`.${arrowName}`).find('Icon').prop('name')).toBe('expand_less')
  })

  it('should focus input after focusing combo box', () => {
    const wrapper = mount(
      <ComboBoxValue
        getToggleButtonProps={props => props}
        getInputProps={() => {}}
        getClearButtonProps={() => {}}
        renderValue={item => item}
      />
    )

    wrapper.simulate('focus')

    const input = wrapper.find(`.${innerName}`).find('input')

    expect(document.activeElement).toBe(input.getDOMNode())
  })

  it('should focus input after clearing combo box', () => {
    const wrapper = mount(
      <ComboBoxValue
        selectedItems={[ 'abc' ]}
        getToggleButtonProps={() => {}}
        getInputProps={() => {}}
        getClearButtonProps={props => props}
        renderValue={item => item}
      />
    )

    wrapper.find(`.${clearName}`).simulate('click')

    wrapper.setProps({ selectedItems: [] })

    // Focus after 'clear' button is delayed
    jest.runAllTimers()

    const input = wrapper.find(`.${innerName}`).find('input')

    expect(document.activeElement).toBe(input.getDOMNode())
  })

  it('should not try to focus with delay when it is already unmounted', () => {
    const spy = jest.fn()

    const wrapper = mount(
      <ComboBoxValue
        selectedItems={[ 'abc' ]}
        getToggleButtonProps={() => {}}
        getInputProps={() => {}}
        getClearButtonProps={props => props}
        renderValue={item => item}
      />
    )

    wrapper.instance().focus = spy

    wrapper.find(`.${clearName}`).simulate('click')

    wrapper.unmount()

    // Focus after 'clear' button is delayed
    jest.runAllTimers()
    expect(spy.mock.calls.length).toBe(0)
  })
})
