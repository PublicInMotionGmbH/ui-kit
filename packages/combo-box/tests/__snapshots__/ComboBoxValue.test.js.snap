// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ComboBoxValue /> should render correctly 1`] = `
<button
  className="talixo-combo-box__value talixo-combo-box__value--placeholder"
>
  <div
    className="talixo-combo-box__inner"
  >
    <input />
  </div>
  <div
    className="talixo-combo-box__controls"
  >
    <span
      className="talixo-combo-box__arrow"
    >
      <Icon
        name="expand_more"
      />
    </span>
  </div>
</button>
`;
